### ISSUE TITLE
Element Interactability checks are not performed on Element click

### ISSUE THREAD

**Issue by AutomatedTester** [w3c/webdriver#1097]
In chapter [14. Element Interaction](https://w3c.github.io/webdriver/webdriver-spec.html#element-click#x14.-element-interaction), section [14.1 Element Click](https://w3c.github.io/webdriver/webdriver-spec.html#element-click#x14.1-element-click):
> If element’s container is still not in view, return error with error code element not interactable. 

The other Element Interaction commands check if an element is interactable before interacting with it

**Comment by andreastt** [comment]
Element Click does check if the element is interactable:

> 5. If element’s container is still not in view, return error with error code element not interactable.
>
> 6. If element’s container is obscured by another element, return error with error code element click intercepted.

**Comment by AutomatedTester** [comment]
@andreastt 

I had put the wrong title, I have updated. The specification does not do a wait for interactable before carrying on. We have this in the other element interaction commands but not click. We are also not in line with some of the legacy drivers from the Selenium project

**Comment by Rajagopalan-M** [comment]
@AutomatedTester  May I know whether they have included the visibility check now?

**Comment by Rajagopalan-M** [comment]
@AutomatedTester  Hi, Is this updated ? Will this facility comes in geckodriver in future?

**Comment by AutomatedTester** [comment]
@Rajagopalan-M when the bug here is done then other drivers can implement it and not before

**Comment by Rajagopalan-M** [comment]
@AutomatedTester  No, I did not understand, What do you mean by other drivers can implement? Which driver?

**Comment by AutomatedTester** [comment]
@Rajagopalan-M this bug is to make sure we specify _how_ to do it. When its done then we can raise a bug against geckodriver to make sure their is code for it

**Comment by Rajagopalan-M** [comment]
@AutomatedTester Okay, thank you.
